test/e2e/gateway_test.go

```diff
@@ -458,6 +479,42 @@ var _ = Describe("Gateway", func() {
 				}, time.Second, 200*time.Millisecond).Should(Equal(404))
 			})
 
+			It("should direct requests that use cluster_header to the proper upstream", func() {
+				// Construct upstream name {{name}}_{{namespace}}
+				us := tu.Upstream
+				upstreamName := translator.UpstreamToClusterName(core.ResourceRef{Namespace: us.Metadata.Namespace, Name: us.Metadata.Name})
```

Comment:
nit, just a bit shorter
```suggestion
				upstreamName := translator.UpstreamToClusterName(us.Metadata.Ref())
```